### ISSUE TITLE
will this work in before hooks and in custom commands?

### ISSUE THREAD

**Issue by bgk17** [bahmutov/cypress-if#51]
I'm trying to detect if I'm logged in and if so, log out in a before hook. I put it in my custom login command. It's failing when it can't find an element (which I would think would just continue to the then())

**Comment by bahmutov** [comment]
I don't think I have enough information to answer this question

**Comment by bgk17** [comment]
can I use cypress-if in my before hook to check if i'm logged in, and if so, log myself out?

**Comment by bahmutov** [comment]
`before` hook runs only once before all tests. You probably want to run it in `beforeEach` hook

**Comment by bgk17** [comment]
Yes, sorry. But that scenario is perfectly fine, right? When I was trying before it was failing to find the element and then doing my .then(), which was confusing me. I will check again.
